### ISSUE TITLE
__init__.py functions being used in submodules

### ISSUE THREAD

**Issue by carsen-stringer** [MouseLand/suite2p#428]
I would like to undo the use of the init files in io, extraction and registration because it obscures where the functions are coming from, can you please take these out when you get a chance? it also makes it more confusing in the GUI, there are bugs in the GUI introduced by this.

I am happy with the init usage in the main package for default_ops and run_s2p

**Comment by nickdelgrosso** [comment]
Hi Carsen,

Can you provide some more details about this? \_\_init\_\_.py is the standard way to create packages in Python, and besides simplifying the usage of packages (by clearly highlighting the public interfaces), it also helps decrease bugs by decoupling the package's internal organization from its usage externally.  If there are import errors, they can be easily fixed, but more info would be a big help.

Best wishes,

Nick

**Comment by marius10p** [comment]
@nickdelgrosso any changes that you make have to remain compatible with the GUI, so please always make sure the GUI can still work. It isn't a huge problem if the GUI doesn't work on the dev branch, but frankly I'm surprised you've been able to test anything at all without checking out the results in the GUI periodically.

**Comment by nickdelgrosso** [comment]
@marius10p Yes, of course we check that the GUI stilll works on a surface level, but because there are no automated tests for the gui code, it is very likely that there will be the occasional bug that gets through.  When this was brought up in our early planning meetings, a testing strategy for the gui was selected: before anything gets merged to master or released on Pypi, @carsen-stringer would test the gui and run an analysis on the newest release, checking that nothing broke there along the way so it could be fixed before it reaches any users.  If our bug tolerance for the gui on the dev branch has changed, then different processes should be followed, as was pointed out during that meeting.  If you would like to discuss testing and merging processes further on the issue board, please open up a new issue on the topic so that this issue can be closed when the bug is resolved.

**Comment by marius10p** [comment]
Sounds good. Please make it a priority to fix the GUI on the dev branch, so I can check out the results of this branch. It appears the master has not been working correctly since April, at least for cell detection, and it might make more sense to merge the dev in if it works correctly.

**Comment by chriski777** [comment]
What are some of the bugs we're seeing on the GUI for the dev branch?

**Comment by marius10p** [comment]
I think Carsen fixed it.

**Comment by carsen-stringer** [comment]
one of the bugs had to do with a key name change which was easy to fix, another had to do with the __main__ file also easy to fix, and another I fixed now had to do with the fact that tiffs weren't being forced to be int16 when written to binary (this won't error with current tests but will with other tiffs like the longer recording). This one was a little harder to find only because the binary file wasn't throwing an out of range error -- I think if we stick to int16 the error handling should work fine :) 

wrt putting functions into the __init__ file that's a great point nick, any outward facing functions can go there, like those used in run_s2p. I think this is largely the case other than in the detection and extraction module where there are functions in the __init__ that aren't used in run_s2p. I'm confused about pc.py though, why is this a separate file from metrics.py? I've moved that back into metrics.py. 

wrt testing the gui, the __main__ function is really the only thing it heavily relies on so changes should not be made there unless you are testing the gui when making those changes. The key errors we can manage as we go.
